### PR TITLE
Issue #281 Add a hook that allows to append sockets to weapons

### PR DESCRIPTION
**Pull request by Musashi1584**

### CODE REVIEW

File: X2WOTCCommunityHighlander/Src/Engine/Classes/SkeletalMeshSocket.uc [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 1998-2011 Epic Games, Inc. All Rights Reserved.
+ */
+class SkeletalMeshSocket extends Object
```
Comment (robojumper) [inline X2WOTCCommunityHighlander/Src/Engine/Classes/SkeletalMeshSocket.uc]: Missing change comments?

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/X2DownloadableContentInfo.uc [hunk 2]
```diff
@@ -475,3 +475,14 @@ static function bool CanWeaponApplyUpgrade(XComGameState_Item WeaponState, X2Wea
 }
 /// End Issue #260
 
+/// Start Issue #281
+/// <summary>
+/// Called from XGWeapon.CreateEntity
+/// Allows DLC/Mods to append sockets to weapons
+/// NOTE: To create new sockets from script you need to unconst SocketName and BoneName in SkeletalMeshSocket
+/// </summary>
+static function DLCAppendWeaponSockets(out array<SkeletalMeshSocket> NewSockets, XComWeapon Weapon, optional XComGameState_Item ItemState = none)
```
Comment (robojumper) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/X2DownloadableContentInfo.uc]: Why is ItemState optional...

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XGWeapon.uc [hunk 3]
```diff
@@ -100,6 +100,10 @@ simulated function Actor CreateEntity(optional XComGameState_Item ItemState=none
 
  		kNewWeapon = Spawn(Template.Class, kOwner,,,,Template);
 		
+		// Start Issue #281
+		DLCInfoAddSockets(kNewWeapon, ItemState);
```
Comment (robojumper) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XGWeapon.uc]: When there's an ``assert(InternalWeaponState != none);` above?

Also, why is it passing `ItemState`, not `InternalItemState`?

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/X2DownloadableContentInfo.uc (hunk 2 above)
Comment (Musashi1584) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/X2DownloadableContentInfo.uc]: cause its optionial in CreateEntity

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XGWeapon.uc (hunk 3 above)
Comment (Musashi1584) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/XGWeapon.uc]: Good point, will change that.

**Pull request merged**